airbyte-cdk/python/unit_tests/sources/streams/http/test_http.py

```diff
@@ -156,6 +156,48 @@ def test_stub_custom_backoff_http_stream(mocker):
     # TODO(davin): Figure out how to assert calls.
 
 
+class AutoFailFalseHttpStream(StubBasicReadHttpStream):
```

Comment:
What about 400, 401, > 200, when we catch success with getting response, but failed do to permissions level or unauthorized?

Is it catches as well?